app/controllers/authenticate_controller.rb

```diff
@@ -222,16 +211,9 @@ def handle_authentication_error(err)
     when Errors::Authentication::Jwt::TokenExpired
       raise Unauthorized.new(err.message, true)
 
-    when Errors::Authentication::OAuth::ProviderDiscoveryTimeout
-      raise GatewayTimeout
```

Comment:
@sashaCher can you please elaborate on this change? why are we not returning a gateway timeout error in this case?
![image](https://user-images.githubusercontent.com/29940926/129565780-bbbb614d-fc29-4deb-8662-d41e1131ea8c.png)

this seems a much better error than the generic 401

Reply:
Updated PR description

Reply:
@sashaCher Can you add the same context to the body of the git commit message?  That, imo, is the right place to capture it so it will live forever and be easily accessible to devs.  What I typically do is write my explanation in a commit message, then just copy that into the main PR description.